pages/contest-list/contest.jsx

```diff
@@ -1,43 +1,59 @@
 import axios from 'axios';
-import { useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import styles from '../../styles/pages/constest-list/contest.module.css';
 
 const MyApp = () => {
-    const [upcoming, setUpcoming] = useState({
-        title: 'hackathon',
-        desc: '',
-        location: 'hackerEarth',
-        time: { start: '', end: '', zone: '' },
-    });
-    const [show, setShow] = useState([
-        {
-            title: 'hackathon',
-            desc: '',
-            location: 'hackerEarth',
-            time: { start: 'fdsfdsf', end: 'fdfds', zone: '' },
-        },
-    ]);
+
+    const allData = React.useRef(null);
+    const [show, setShow] = useState(null);
     const [status, setStatus] = useState(false);
     const [sites, setSites] = useState(false);
     const [time, setTime] = useState(false);
+    const siteTag = {
+        hackerEarth: "hackerearth.com",
+        hackerRank:"hackerrank.com",
+        codeForces:"codeforces.com",
+        leetcode:"leetcode.com",
+        geeksForGeeks:"gfg.com",
+    }
+
+    const [filterData , setFilterData]=useState(
+        {
+                [siteTag.hackerEarth]:false,
+                [siteTag.hackerRank]:false,
+                [siteTag.codeForces]:false,
+                [siteTag.leetcode]:false,
+                [siteTag.geeksForGeeks]:false,
+        }
+    )
+
+    const [active, setActive] = useState(''); 
 
-    const [active, setActive] = useState('');
-    const url = 'https://contests-api.herokuapp.com/upcoming';
     useEffect(() => {
+        const url = 'https://contests-api.herokuapp.com/upcoming';
         axios
             .get(url)
-            .then((Response) => {
-                console.log(Response.data.data);
-                setUpcoming(Response.data.data);
-                setShow(Response.data.data);
+            .then((res) => {
+                allData.current = res.data.data;
+                setShow(res.data.data);
+                console.log(res)
             })
             .catch((err) => {
                 console.log(err);
             })
-            .then(() => {
-                console.log('request send');
-            });
     }, []);
+
+    useEffect(()=>{
+        console.log(Object.entries)
+        if(Object.entries(filterData).filter((e) => e[1]).length === 0) {
+            setShow(allData.current)
+        }else{
+            setShow(allData.current.filter((data)=> 
```

Comment:
if their is something true in filterdata then it will filter the api data accordingly from allData